### PR TITLE
RUMM-323 `macOS/Catalyst` support dropped

### PR DESCRIPTION
**Pull request by buranmert**
### What and why?

Although supporting `macOS/Catalyst` is a good thing, it can also add complexity to codebase in the form of compiler directive checks (Eg: `#if (iOS)... #else ...`).

Given that our users currently do not widely use the SDK in those platforms, we decided to drop support for them.

And we keep the door open: in case that `macOS/Catalyst` support is demanded, we will work on it.

### How?

`macOS/Catalyst` support is removed from

1. pbxproj files (`Datadog.pbxproj`)
2. dependency manager spec files (`podspec`, `Package.swift`)
3. dependency manager tests (`dependency-manager-tests/spm/Package.swift`)

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

File: Datadog.podspec [hunk 1]
```diff
@@ -11,7 +11,6 @@ Pod::Spec.new do |s|
 
   s.swift_version      = '5.1'
```
Comment (ncreated) [inline Datadog.podspec]: We also need to update `s.summary` so it doesn't mention `macOS`.

File: DatadogObjc.podspec [hunk 2]
```diff
@@ -11,7 +11,6 @@ Pod::Spec.new do |s|
 
   s.swift_version      = '5.1'
```
Comment (ncreated) [inline DatadogObjc.podspec]: Same here - let's update `s.summary` ☝️.

**Comment by buranmert** [comment]
@ncreated I made changes ✅ I will push it once I get `SPMTestProject` green again ⏳

File: dependency-manager-tests/spm/Makefile [hunk 3]
```diff
@@ -6,7 +6,9 @@ endif
 
 test:
 		@echo "⚙️  Testing SPM for remote branch: '${GIT_BRANCH}'..."
-		@sed "s|REMOTE_GIT_BRANCH|${GIT_BRANCH}|g" Package.swift.src > Package.swift
-		swift package update
-		swift build
+		@sed -i "" "s|branch = *.;|branch = ${GIT_BRANCH};|g" SPMProject.xcodeproj/project.pbxproj
```
Comment (ncreated) [inline dependency-manager-tests/spm/Makefile]: 👏 💪

File: dependency-manager-tests/spm/SPMProject/AppDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,86 @@
+//
+//  AppDelegate.swift
+//  SPMProject
+//
```
Comment (ncreated) [inline dependency-manager-tests/spm/SPMProject/AppDelegate.swift]: Please fix the license header. IDK why `./tools/license/check-license.sh` did pass on CI even though it captured a file with no license:

Screenshot 2020-03-26 at 16 18 04

But no failure on CI is a separate issue, I'll put this into JIRA.

File: dependency-manager-tests/spm/SPMProject/AppDelegate.swift [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+import UIKit
+import CoreData
+
+import Datadog
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+
```
Comment (ncreated) [inline dependency-manager-tests/spm/SPMProject/AppDelegate.swift]: Let's add `SPMProject` folder to both `sources.swiftlint.yml` and `tests.swiftlint.yml` and align the template syntax with our convention. Also let's add the build phase to keep notified when updating the project.

```yml
# tools/linter/sources.swiftlint.yml

included:
  - Sources
  - instrumented-tests/Integration/Integration
  - instrumented-tests/Benchmark/Benchmark
  - instrumented-tests/http-server-mock/Sources
  - dependency-manager-tests/spm/SPMProject 👈
```
```yml
# tools/linter/tests.swiftlint.yml

included:
  - Tests
  - instrumented-tests/Integration/IntegrationTests
  - instrumented-tests/Benchmark/BenchmarkTests
  - instrumented-tests/http-server-mock/Tests
  - dependency-manager-tests/spm/SPMProject/Tests 👈
```

File: dependency-manager-tests/spm/SPMProject/DetailViewController.swift [hunk 6]
```diff
@@ -0,0 +1,40 @@
+//
+//  DetailViewController.swift
+//  SPMProject
+//
```
Comment (ncreated) [inline dependency-manager-tests/spm/SPMProject/DetailViewController.swift]: Same here - license header. Also, please run `./tools/license/check-license.sh` to see if all is good.

File: dependency-manager-tests/spm/SPMProject/MasterViewController.swift [hunk 7]
```diff
@@ -0,0 +1,194 @@
+//
+//  MasterViewController.swift
+//  SPMProject
+//
```
Comment (ncreated) [inline dependency-manager-tests/spm/SPMProject/MasterViewController.swift]: Same here - license header. Also, please run `./tools/license/check-license.sh` to see if all is good.

File: dependency-manager-tests/spm/SPMProject/MasterViewController.swift [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+//  Created by Mert Buran on 26/03/2020.
+//  Copyright © 2020 Datadog. All rights reserved.
+//
+
+import UIKit
+import CoreData
```
Comment (ncreated) [inline dependency-manager-tests/spm/SPMProject/MasterViewController.swift]: CoreData 💔. How about using "Single View App" with just `UITests` enabled and no Core Data nor unit tests, as we won't need them?

File: dependency-manager-tests/spm/SPMProject.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -0,0 +1,637 @@
...
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CODE_SIGN_STYLE = Automatic;
+				INFOPLIST_FILE = SPMProject/Info.plist;
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
+				PRODUCT_BUNDLE_IDENTIFIER = DD.SPMProject;
```
Comment (ncreated) [inline dependency-manager-tests/spm/SPMProject.xcodeproj/project.pbxproj]: `com.datadogqh.SPMProject` would keep it more consistent with other bundle identifiers in the repository 🙂.

File: dependency-manager-tests/spm/SPMProject/MasterViewController.swift (hunk 8 above)
Comment (buranmert) [reply dependency-manager-tests/spm/SPMProject/MasterViewController.swift]: I feel you 😄 
I kept it checked at project creation in order to make the project more realistic, it doesn't seem to cause any problems so far.
Adding unit tests afterwards usually cause me pain, that's why I kept it in although as you said we don't use them for now.

To be honest, I didn't pay attention to details as we are planning to change the project structure soon. 
Shall I make these changes now or leave for later?

Comment (ncreated) [reply dependency-manager-tests/spm/SPMProject/MasterViewController.swift]: Sure, we can tackle it later when doing `UITests` for SPM project 👌🙂.

**Pull request merged**